src/app/_components/Search/Search.tsx

```diff
@@ -0,0 +1,447 @@
+import {
+  advancedSearchConfig,
+  advancedSearchKeywords,
+  useSearchPageUrl,
+} from '../../../common/queries/useSearchQuery';
+import { useAppDispatch, useAppSelector } from '../../../common/state/hooks';
+import {
+  blur,
+  clearSearchTerm,
+  focus,
+  selectIsSearchFieldFocused,
+  selectSearchPreviewScrollLeft,
+  selectTempSearchTerm,
+  setSearchPreviewScrollLeft,
+  setSearchTerm,
+  setTempSearchTerm,
+} from '../../../features/search/search-slice';
+import { Button } from '../../../ui/Button';
+import { Input, InputProps } from '../../../ui/Input';
+import { Kbd } from '../../../ui/Kbd';
+import { Text } from '../../../ui/Text';
+import { DoubleGradientBorderWrapper } from './DoubleGradientBorderWrapper';
+import {
+  BlockResultItem,
+  BnsResultItem,
+  CoinbaseResultItem,
+  ContractCallResultItem,
+  ContractDeployResultItem,
+  ResultItem,
+  TenureChangeResultItem,
+  TokenTransferResultItem,
+} from './RestultItem';
+import { SearchLink } from './SearchLink';
+
+export function SearchResultsWrapper({ children }: { children: ReactNode }) {
+  return (
+    <Flex
+      zIndex={-1}
+      flexDirection={'column'}
+      position={'absolute'}
+      background={'surfaceSecondary'}
+      left={'calc(-1 * var(--stacks-spacing-3))'}
+      right={'calc(-1 * var(--stacks-spacing-3))'}
+      top={'calc(-1 * var(--stacks-spacing-3))'}
+      borderRadius={'1.125rem'}
+      boxShadow={'0px 16px 32px 0px rgba(184, 180, 176, 0.20)'}
+      gap={4}
+    >
+      {children}
+    </Flex>
+  );
+}
+
+function KeywordsPreview() {
+  return (
+    <Flex rowGap={1} columnGap={2} flexWrap={'wrap'} mt={1}>
+      {Object.keys(advancedSearchConfig).map(key => {
+        if (key === 'TERM:') return null;
+        return (
+          <Flex
+            key={key}
+            fontSize={'xs'}
+            bg="surfaceHighlight"
+            borderRadius="md"
+            color="textSubdued"
+            whiteSpace="pre"
+            textTransform={'uppercase'}
+            fontFamily="var(--font-matter-mono)"
+            fontWeight={'400'}
+          >
+            <Text color={'textSecondary'}>{key}</Text>
+            <Text color={'textTertiary'}>({advancedSearchConfig[key].type})</Text>
+          </Flex>
+        );
+      })}
+    </Flex>
+  );
+}
+
+function DocsLink() {
+  return (
+    <SearchLink gap={1} lineHeight={'base'}>
+      <Icon w={4} h={4}>
+        <Info />
+      </Icon>
+      <span>Learn more about search syntax</span>
+    </SearchLink>
+  );
+}
+
+function QuickLinks() {
+  return (
+    <Flex
+      flexDirection={'column'}
+      backgroundColor={'surfacePrimary'}
+      p={4}
+      fontFamily="var(--font-instrument-sans)"
+      gap={2}
+      borderBottomLeftRadius={'1.125rem'}
+      borderBottomRightRadius={'1.125rem'}
+    >
+      <Text fontSize={'xs'} color={'textSecondary'} lineHeight={'base'}>
+        Quick links
+      </Text>
+      <Flex gap={2}>
+        <Button variant={'text'} size={'sm'}>
```

Comment:
I think all these buttons should just use an arary of the button names and map them to the react component. Also, this button with the right arrow is in the design system. I have added a lot of the button design system in my navbar PR but I didn't add the designs for buttons with arrows. You could wait until my PR gets merged then rebase, then add this variant. I think we should just modify the button component to accept a "withArrow" prop and use that to change the jsx